library/src/main/java/com/fidesmo/fdsm/FidesmoApiClient.java

```diff
@@ -245,13 +247,36 @@ public static String lamei18n(JsonNode n) {
         // For missing values
         if (n == null)
             return "";
-        if (n.size() > 0) {
-            Map<String, Object> langs = mapper.convertValue(n, new TypeReference<Map<String, Object>>() {
-            });
-            Map.Entry<String, Object> first = langs.entrySet().iterator().next();
-            return langs.getOrDefault(Locale.getDefault().getLanguage(), langs.getOrDefault("en", first.getValue())).toString();
+        if (!n.isEmpty()) {
+            boolean isNewFormat = n.has("text");
+            if (isNewFormat) {
+                return FidesmoApiClient.buildMessageWithParams(n);
+            }
+            else {
+                Map<String, Object> langs = mapper.convertValue(n, new TypeReference<Map<String, Object>>() {
+                });
+                Map.Entry<String, Object> first = langs.entrySet().iterator().next();
+                return langs.getOrDefault(Locale.getDefault().getLanguage(), langs.getOrDefault("en", first.getValue())).toString();
+            }
         } else {
             return n.asText();
         }
     }
+
+
+    public static String buildMessageWithParams(JsonNode n) {
+
+        String text =  n.path("text").asText();
+
+        JsonNode paramsNode = n.path("params");
```

Comment:
So if params array is not present does it return an empty one or null? As it's an optional field so I'm afraid can fail here

Reply:
One of the tests (`returnMessagesInNewFormatWithoutParams()`) covers that case and it does not fail.
I can another one for the case when params is empty

Reply:
OK then, thanks!